### PR TITLE
Add CGroupResourceLimits class.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

This PR is related to issue #22444

### CODE REVIEW

File: vespalib/src/vespa/vespalib/util/cgroup_resource_limits.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#include <cstdint>
+#include <functional>
+#include <optional>
+#include <string>
+#include <map>
+
+namespace vespalib {
+
+/*
+ * Class for getting cgroup resource limits.
```
Comment (geirst) [inline vespalib/src/vespa/vespalib/util/cgroup_resource_limits.h:14]: Please add some details on which cgroup setups that are supported.

File: vespalib/src/vespa/vespalib/util/resource_limits.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include <cstdint>
+
+namespace vespalib {
+
+/*
+ * Class for getting resource limits.
```
Comment (geirst) [inline vespalib/src/vespa/vespalib/util/resource_limits.h:10]: Please add some details on how the resource limits are sampled.

**Review by geirst — APPROVED** [review]
👍

File: vespalib/src/vespa/vespalib/util/resource_limits.h (hunk 2 above)
Comment (toregge) [reply vespalib/src/vespa/vespalib/util/resource_limits.h]: Done.

File: vespalib/src/vespa/vespalib/util/cgroup_resource_limits.h (hunk 1 above)
Comment (toregge) [reply vespalib/src/vespa/vespalib/util/cgroup_resource_limits.h]: Done.

**Review by geirst — APPROVED** [review]
Perfect 👍

**Pull request merged**